sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/DirectoryAPITest.groovy

```diff
@@ -3001,4 +3000,29 @@ class DirectoryAPITest extends APISpec {
         then:
         thrown(DataLakeStorageException)
     }
+
+    @Unroll
+    def "Get file and subdirectory client"() {
+        setup:
+        dc = fsc.getDirectoryClient(resourcePrefix +  generatePathName())
+
+        when:
+        dc.getFileClient(subResourcePrefix + generatePathName())
+
+        then:
+        notThrown(IllegalArgumentException)
+
+        when:
+        dc.getSubdirectoryClient(subResourcePrefix + generatePathName())
+
+        then:
+        notThrown(IllegalArgumentException)
+
+        where:
+        resourcePrefix          | subResourcePrefix         || _
+        ""                      | ""                        || _
+        Utility.urlEncode("%")  | ""                        || _ // Resource has special character
+        ""                      | Utility.urlEncode("%")    || _ // Sub resource has special character
```

Comment:
I think we should check the path against the expected string to ensure the right amount of encoding/decoding is happening

Reply:
I can add that